libs/langchain/langchain/document_loaders/s3_file.py

```diff
@@ -35,4 +35,8 @@ def load(self) -> List[Document]:
             os.makedirs(os.path.dirname(file_path), exist_ok=True)
             s3.download_file(self.bucket, self.key, file_path)
             loader = UnstructuredFileLoader(file_path)
-            return loader.load()
+            docs = loader.load()
+            for doc in docs:
```

Comment:
Sure.

Reply:
Done.
I also changed the way to load document by inheriting `UnstructuredBaseLoader` and not using `UnstructuredFileLoader` anymore. I think it's cleaner this way. WDYT ?

Reply:
You mean we pass the class of parser and in load() we instantiate it with the file name as constructor arg ?

Reply:
I think that would be a bit better, though maybe not critical since the current change isn't breaking for users.

At a higher level, I'm trying to decouple fetching from content and rely more on composition to make it as easy as possible to re-use existing parsers on files regardless of where the files are stored. 

We have a BlobGenerator abstraction that could use an S3 implementation that is able to fetch file blobs from s3 that match certain criteria.

Reply:
Let me know if you want to make changes now or if you want me to merge as is